addons_installer.FCMacro

```diff
@@ -108,24 +107,40 @@ class AddonsInstaller(QtGui.QDialog):
         self.verticalLayout.addLayout(self.horizontalLayout)
 
         self.retranslateUi()
-        
+
         QtCore.QObject.connect(self.buttonCancel, QtCore.SIGNAL("clicked()"), self.reject)
         QtCore.QObject.connect(self.buttonInstall, QtCore.SIGNAL("clicked()"), self.install)
         QtCore.QObject.connect(self.buttonRemove, QtCore.SIGNAL("clicked()"), self.remove)
         QtCore.QObject.connect(self.labelDescription, QtCore.SIGNAL("linkActivated(QString)"), self.showlink)
         QtCore.QObject.connect(self.listAddons, QtCore.SIGNAL("currentRowChanged(int)"), self.show)
         QtCore.QMetaObject.connectSlotsByName(self)
         self.update()
-        
+
     def retranslateUi(self):
-        self.setWindowTitle(QtGui.QApplication.translate("AddonsInstaller", "Addons installer", None, QtGui.QApplication.UnicodeUTF8))
-        self.labelDescription.setText(QtGui.QApplication.translate("AddonsInstaller", "Downloading addons list...", None, QtGui.QApplication.UnicodeUTF8))
-        self.buttonCancel.setText(QtGui.QApplication.translate("AddonsInstaller", "Close", None, QtGui.QApplication.UnicodeUTF8))
-        self.buttonInstall.setText(QtGui.QApplication.translate("AddonsInstaller", "Install / update", None, QtGui.QApplication.UnicodeUTF8))
-        self.buttonRemove.setText(QtGui.QApplication.translate("AddonsInstaller", "Remove", None, QtGui.QApplication.UnicodeUTF8))
-        
+        self.setWindowTitle(
+            QtGui.QApplication.translate("AddonsInstaller", "Addons installer", None, QtGui.QApplication.UnicodeUTF8))
+        self.labelDescription.setText(
+            QtGui.QApplication.translate("AddonsInstaller", "Downloading addons list...", None,
+                                         QtGui.QApplication.UnicodeUTF8))
+        self.buttonCancel.setText(
+            QtGui.QApplication.translate("AddonsInstaller", "Close", None, QtGui.QApplication.UnicodeUTF8))
+        self.buttonInstall.setText(
+            QtGui.QApplication.translate("AddonsInstaller", "Install / update", None, QtGui.QApplication.UnicodeUTF8))
+        self.buttonRemove.setText(
+            QtGui.QApplication.translate("AddonsInstaller", "Remove", None, QtGui.QApplication.UnicodeUTF8))
+
+    def update_repos(self, repos=[]):
+        if not (repos == []):
+            self.repos = repos
+        else:
+            return self.repos
+
     def update(self):
-        self.update_worker = UpdateWorker()
+        self.info_worker = InfoWorker()
+        self.info_worker.addon_repos.connect(self.update_repos)
+        # self.info_worker.start()
+
+        self.update_worker = UpdateWorker(self.info_worker)
```

Comment:
Setup "InfoWorker" which fetches the repo information in the background, preventing crashes while improving use expirience